autopilot/prefattach.go

```diff
@@ -127,8 +132,9 @@ func (p *PrefAttachment) NodeScores(g ChannelGraph, chans []Channel,
 		}
 
 		// Otherwise we score the node according to its fraction of
-		// channels in the graph.
-		score := float64(nodeChans) / float64(graphChans)
+		// channels in the graph, scaled such that the highest-degree
+		// node will be given a score of 1.0.
+		score := float64(nodeChans) / float64(maxChans)
```

Comment:
This change significantly alters the existing algorithm. I don't see why this change is needed, since without this, the score is _still_ in the range of 0 to 1.

Reply:
The idea here is that it will make it easier to combine multiple heuristics. I ran into issues when attempting to combine the scores returned from the existing algorithm with the bos scores. 

Before this change the `prefattach` scores would look like `[A: 0.002, B: 0.03, C: 0.0024, ...]`, i.e. very small scores for most nodes. The `bos` scores would be binary in the sense that nodes not in the list would implicitly be given a zero-score: `[A: 1.0, C: 1.0, ...]`. 

This made it very hard to find the correct weighting if I wanted to combine them. For instance if I wanted to consider the two heuristics equally, I would give them both a weight of `0.5`. However, for the example above this would result in scores like `[A: 0.5001, B: 0.015, C: 0.5012]`. Since the scales of the scores would be vastly different, to find the right weighting was hard. Also nodes considered good by the `prefattach` algorithm would almost never get picked, since they wasn't found in the score list.

To your first point: this _shouldn't_ change the algorithm, only the scale of the returned scores. Consider the old version for score given to node `i`:
```
score_i := nodeChans_i / graphChans
```
This change scales all these scores by finding the maximum score and multiplying all by `(1.0 / score_max)`:
```
score_max := maxChans / graphChans

score_i = score_i * (1.0 / score_max)
             = (nodeChans_i / graphChans) * (1.0 / (maxChans / graphChans))
             = (nodeChans_i / graphChans) * (graphChans / maxChans)
             = nodeChans_i / maxChans

```

Reply:
> Also nodes considered good by the prefattach algorithm would almost never get picked, since they wasn't found in the score list.

That isn't necessarily a bad thing, they could be node that _seem_ well connected, but in reality aren't able to properly route payment.